### PR TITLE
Alsa: add alsa-tools, add option to load firmware on boot

### PR DESCRIPTION
**Pull request by mstrm**
The firmware in alsa-firmware won't get loaded unless the firmware path is added to hardware.firmware (at least this is what I had to do for my rme hdsp card), so I added an option for it. I also added alsa-tools and pyalsa.

### CODE REVIEW

File: nixos/modules/services/audio/alsa.nix [hunk 1]
```diff
@@ -80,6 +88,9 @@ in
         };
       };
 
+    hardware.firmware = mkIf config.sound.enableAlsaFirmware
+      [ "${pkgs.alsa-firmware}/lib/firmware" ];
+
```
Comment (bencef) [inline nixos/modules/services/audio/alsa.nix]: first, thanks for doing this.
second, I'm a nix noobie but shouldn't you also add it to systemPackages here?

Comment (mstrm) [inline nixos/modules/services/audio/alsa.nix]: I tried it and it gets added automatically.

Comment (bencef) [inline nixos/modules/services/audio/alsa.nix]: Ok.

File: pkgs/os-specific/linux/alsa-tools/default.nix [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    sed -i "s,/bin/lo10k1,$out/bin/lo10k1,g" $out/bin/init_{audigy,audigy_eq10,live}
+    sed -i "s,bindir=/bin,bindir=$out/bin," $out/bin/init_live
+    wrapProgram $out/bin/hwmixvolume --prefix PYTHONPATH : $PYTHONPATH
+    mv $out/usr/local/bin/sbiload $out/bin/sbiload
+    mkdir -p $out/share/sounds
+    mv $out/usr/local/share/sounds/opl3 $out/share/sounds/opl3
+    rm -r $out/usr
+  '';
+
+  meta = {
```
Comment (the-kenny) [inline pkgs/os-specific/linux/alsa-tools/default.nix]: Please add meta.license.

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -8589,6 +8591,8 @@ let
 
   alsa-firmware = callPackage ../os-specific/linux/alsa-firmware { };
 
+  alsaTools = callPackage ../os-specific/linux/alsa-tools { };
```
Comment (the-kenny) [inline pkgs/top-level/all-packages.nix]: no need to use camelCase for attributes anymore. If the package name is ```also-tools```, please use that name for the attribute.

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -8094,6 +8094,22 @@ let
...
+  pyalsa = buildPythonPackage (rec {
+    name = "pyalsa-1.0.29";
+    disabled = isPy3k;
+    buildInputs = [ pkgs.alsaLib ];
+    src = pkgs.fetchurl {
+      url = "ftp://ftp.alsa-project.org/pub/pyalsa/pyalsa-1.0.29.tar.bz2";
+      sha256 = "87ea6d8a2b7a9d7b015cdd84c898dc5e524f770ae6795e0d32ac2234311c953a";
+    };
+
+    meta = {
```
Comment (the-kenny) [inline pkgs/top-level/python-packages.nix]: meta.license is missing here too.

File: pkgs/os-specific/linux/alsa-tools/default.nix [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    # Dont build any of the firmware loader programs. I found no easy way to
+    # modify the firmware search path. Instead firmware can be loaded by setting
+    # sound.enableAlsaFirmware to true and rebooting.
+    sed -i 's,[^ ]*loader,,g' Makefile
+
+    sed -i 's,"$CONFIGURE_ARGS","$(CONFIGURE_ARGS)",' Makefile
+  '';
+
+  buildPhase = "make configure"; # yes this builds the project too
+  installPhase = "make install";
```
Comment (the-kenny) [inline pkgs/os-specific/linux/alsa-tools/default.nix]: is the override for ```installPhase``` really necessary?

File: pkgs/os-specific/linux/alsa-tools/default.nix [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  buildInputs = [ autoconf automake alsaLib pkgconfig gtk2 gtk3 fltk13 qt3 perl python pygobject pygtk pyalsa makeWrapper ];
+
+  CONFIGURE_ARGS="--prefix=";
+  DESTDIR="$(out)";
+  AUTOMAKE_DIR="${automake}/share/automake-1.12";
+
+  phases = [ "unpackPhase" "patchPhase" "buildPhase" "installPhase" "fixupPhase" ];
+
+  patchPhase = ''
+    find . -name gitcompile -exec sed -i 's,#!/bin/sh,#!${stdenv.shell},' {} \;
```
Comment (the-kenny) [inline pkgs/os-specific/linux/alsa-tools/default.nix]: ```patchShebangs``` should work here.

**Comment by mstrm** [comment]
@the-kenny: I fixed all the issues you mentioned.

**Comment by mstrm** [comment]
Cleaned up history.

**Comment by mstrm** [comment]
Fixed firmware loader programs in alsa-tools. Rebased.

**Pull request closed without merging**